docs/spec.md

```diff
@@ -160,10 +163,50 @@ following constraints:
 * (C12) If `quantization_dimension` is not empty, then
   `0 <= quantization_dimension`.
 
-At the moment, `QuantizationScale` is a floating-point constant, but there is
-strong interest in integer-based scales, represented with multipliers and
-shifts. We are planning to explore this in the near future
-([#1404](https://github.com/openxla/stablehlo/issues/1404)).
+In order to allow operation using only integer arithmetic, the floating-point
+`scale`, `S`, is realized using integer `multipler`, `M` and `shift` value,
+`n >= 0`, such that `round_nearest_even(S * 2^n) = M`.
+
+The following demonstrates, using C++ code, a possible implementation of
+deriving the integer parameters `M` and `n` of type `i32` from a
+floating-point scale `S` of type `f64`:
+
+```c++
+    int32_t shift;
+    double mantissa = std::frexp(S, &shift);
+    auto shiftedMantissa = static_cast<int64_t>(std::round(mantissa * (int64_t(1) << 31)));
```

Comment:
General design question: is the constant 31 general to all ops?  Or, different ops (e.g. add vs. multiply) may want different fractional bits?

Reply:
Thanks for bring this up @loganchien. 

>>  different ops (e.g. add vs. multiply) may want different fractional bits?

I think the scaling bit-width of an operation depends on
 - operation itself (so not same constant 31 for all ops)
 - storage bitwidth of the output quantized type
 - implementation. 
 
 With that I temped to think that the bit-scaling to use for an individual  operation as "implementation defined."

@loganchien  @dominicsymes  What do you think?

Reply:
I think there will be different conversions from these quantization parameters to the rescale parameters. For example to 16-bit or 32-bit rescale. However, I also think it would be useful to provide example/reference conversions as we are discussing above for dot product and for add operation.